test/acceptance/resources/apps/sbo-generic-test-app/file-permissions.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+urlencode() {
+    # urlencode <string>
+    local length="${#1}"
+    for (( i = 0; i < length; i++ )); do
+        local c="${1:i:1}"
+        case $c in
+            [a-zA-Z0-9.~_-]) printf "$c" ;;
+            *) printf '%%%02X' "'$c" ;;
+        esac
+    done
+}
+
+echo "Content-Type: text/plain"
+
+[ -z "$SERVICE_BINDING_ROOT" ] && SERVICE_BINDING_ROOT=$(jq -r '.SERVICE_BINDING_ROOT // empty' /tmp/env.json)
+
+[ -z "$SERVICE_BINDING_ROOT" ] && echo -e "Status: 404 SERVICE_BINDING_ROOT Not Found\n" && exit
+
+for f in $(find $SERVICE_BINDING_ROOT -name testfile -type f); do
```

Comment:
@baijum please do not use hardcoded file such as `testfile`. We want to keep the app generic and so to check permissions of any file in the app container.

Rather, make the file path an API parameter, like `/file-permissions/<filepath>` and then test the `<filepath>` instead of the hardcoded `testfile`.

Currently, in case the `testfile` is not part of the binding (so it is not found) and the mounted files have incorrect permissions (for whatever reason), the API returns `OK` and the respective test PASSes even if it should fail, which is incorrect.

Thx

Reply:
I have fixed these issues.